### PR TITLE
Working example

### PR DESCRIPTION
**Pull request by guilhermebodin**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,15 +2,15 @@ language: julia
...
   - 1.0
-  - 1.1
-  - 1.2
-  - 1.3
+  # - 1.1
+  # - 1.2
+  # - 1.3
   - 1.4
-  - nightly
+  # - nightly
```
Comment (raphaelsaavedra) [inline .travis.yml:9]: Is there some reason not to test on nightly while allowing to fail?

File: src/filters/scalar_kalman_filter.jl [hunk 2]
```diff
@@ -113,30 +113,34 @@ end
...
-        scalar_update_v!(kalman_state, y, Z, d)
-        scalar_update_F!(kalman_state, Z, H)
-        scalar_update_att!(kalman_state, Z)
-        scalar_update_a!(kalman_state, T, c)
-        scalar_update_Ptt!(kalman_state, Z)
-        scalar_update_P!(kalman_state, T, RQR)
-        scalar_check_steady_state(kalman_state, tol)
-        if t > skip_llk_instants
-            update_llk!(kalman_state)
+        if kalman_state.steady_state
```
Comment (raphaelsaavedra) [inline src/filters/scalar_kalman_filter.jl:126]: This would look better as an `elseif`, I believe?

File: src/filters/univariate_kalman_filter.jl [hunk 3]
```diff
@@ -143,30 +143,33 @@ end
...
-        update_v!(kalman_state, y, Z, d)
-        update_F!(kalman_state, Z, H)
-        update_att!(kalman_state, Z)
-        update_a!(kalman_state, T, c)
-        update_Ptt!(kalman_state)
-        update_P!(kalman_state, T, RQR)
-        check_steady_state(kalman_state, tol)
-        if t > skip_llk_instants
-            update_llk!(kalman_state)
+        if kalman_state.steady_state
```
Comment (raphaelsaavedra) [inline src/filters/univariate_kalman_filter.jl:155]: Same thing here

File: src/models/locallevel.jl [hunk 4]
```diff
@@ -44,37 +44,43 @@ struct LocalLevel{Fl <: Real} <: StateSpaceModel
...
-        "sigma2_ε" => var(model.system.y),
-        "sigma2_η" => var(model.system.y)
+        "sigma2_ε" => observed_variance,
+        "sigma2_η" => observed_variance
     )
     set_initial_hyperparameters!(model, initial_hyperparameters)
     return
 end
-function constraint_hyperparameters!(model::LocalLevel{Fl}) where Fl
+function constraint_hyperparameters!(model::LocalLevel)
```
Comment (raphaelsaavedra) [inline src/models/locallevel.jl:69]: should probably be called `constrain_hyperparameters!`

File: src/models/locallevel.jl [hunk 5]
```diff
@@ -44,37 +44,43 @@ struct LocalLevel{Fl <: Real} <: StateSpaceModel
...
     constrain_variance(model, "sigma2_ε")
     constrain_variance(model, "sigma2_η")
     return
 end
-function unconstraint_hyperparameters!(model::LocalLevel{Fl}) where Fl
+function unconstraint_hyperparameters!(model::LocalLevel)
```
Comment (raphaelsaavedra) [inline src/models/locallevel.jl:74]: same as above

File: src/smoothers/kalman_smoother.jl [hunk 6]
```diff
@@ -28,9 +28,9 @@ function kalman_smoother!(smoother_output::SmootherOutput,
     r[end] = zeros(Fl, m)
 
     @inbounds for t = n:-1:2
-        if false # Decide how to deal with missing observations
-            # r[t-1, :]     = T' * r[t, :]
-            # N[:, :, t-1]  = T' * N[:, :, t] * T
+        if any(isnan.(system.y[t, :])) #TODO check this
```
Comment (raphaelsaavedra) [inline src/smoothers/kalman_smoother.jl:31]: this seems fine

File: src/smoothers/kalman_smoother.jl [hunk 7]
```diff
@@ -43,9 +43,9 @@ function kalman_smoother!(smoother_output::SmootherOutput,
     end
 
     # Last iteration
-    if false # Decide how to deal with missing observations
-        # r0  = T' * r[1, :]
-        # N0  = T' * N[:, :, 1] * T
+    if any(isnan.(system.y[1, :])) # Decide how to deal with missing observations
```
Comment (raphaelsaavedra) [inline src/smoothers/kalman_smoother.jl:46]: fix comment here

File: test/models/locallevel.jl [hunk 8]
```diff
@@ -33,4 +33,21 @@
...
+    @time filter = kalman_filter(model)
+    @time smoother = kalman_smoother(model)
+    @test filter.Ptt[end] == smoother.V[end] # by construction
+
+    for t in 2:length(model.system.y) - 1
+        @test filter.Ptt[t][1] > smoother.V[t][1] # by construction
+    end
+
+    # forecasting
+    forec = forecast(model, 10)
```
Comment (raphaelsaavedra) [inline test/models/locallevel.jl:52]: could add at least one simple test on the forecasts, for example check if the variance/CIs are growing in each step

File: .travis.yml (hunk 1 above)
Comment (guilhermebodin) [reply .travis.yml:9]: i could not manage the travis quite well, allowing it to fail was not triggering the documentation so I postponed the full set of tests

File: src/smoothers/kalman_smoother.jl (hunk 6 above)
Comment (guilhermebodin) [reply src/smoothers/kalman_smoother.jl:31]: This is not optimized and I wonder if there is another criteria for checking the missing observations

**Pull request merged**